pkg/operator/operator.go

```diff
@@ -223,11 +228,27 @@ func (o *Operator) WithControllers(ctx context.Context, controllers ...controlle
 	return o
 }
 
-func (o *Operator) WithWebhooks(ctx context.Context, ctors ...knativeinjection.ControllerConstructor) *Operator {
+func (o *Operator) WithWebhooks(ctx context.Context, ctors ...knativeinjection.NamedControllerConstructor) *Operator {
 	if !options.FromContext(ctx).DisableWebhook {
 		o.webhooks = append(o.webhooks, ctors...)
 		lo.Must0(o.Manager.AddReadyzCheck("webhooks", webhooks.HealthProbe(ctx)))
 		lo.Must0(o.Manager.AddHealthzCheck("webhooks", webhooks.HealthProbe(ctx)))
+
+		kCtx, _ := knativeinjection.EnableInjectionOrDie(ctx, o.GetConfig())
+		kCtx = injection.WithClient(kCtx, o.GetClient())
+		kCtx = injection.WithNodeClasses(kCtx, []schema.GroupVersionKind{})
+
+		// Add webhook OwnerReference controller (to clean-up owner references added by knative)
+		// for any Validating or Mutating WebhookConfigurations
+		for _, ctor := range ctors {
```

Comment:
Are we getting to a point where we need to think about the number of controllers we have? we're adding in 3 controllers, all each for one resource in the cluster. 

Also as I think more, does it make sense that we have this implemented as 3 genericized controllers? We're only caring about three specific resources in the cluster from a previous installation, right? Can we just do this on startup?

Reply:
Not sure if I'm following -- this really doesn't do much from a goroutine perspective once it's done with its task. This is also only present on the pre-v1.0 versions so we aren't carrying it forward -- also, you get leader election for free and don't have to deal with retry behavior like you would if you had to create your own thread to do the task for you